sycl/doc/extensions/supported/sycl_ext_intel_device_info.md

```diff
@@ -347,3 +348,41 @@ Then the maximum memory bandwidth can be obtained using the standard get\_info()
     if (dev.has(aspect::ext_intel_max_mem_bandwidth)) {
       auto maxBW = dev.get_info<info::device::ext_intel_max_mem_bandwidth>();
     }
+
+# Free Memory #
+
+A new device descriptor will be added which will provide the number of bytes of free memory for the device.
+
+This new device descriptor is only available for devices in the Level Zero platform, and the matching aspect is only true for those devices. The DPC++ default behavior is to expose GPU devices through the Level Zero platform. NOTE: one may need to set
+ZES_ENABLE_SYSMAN=1 to fully enable this extension.
+
+
+## Version ##
+
+The extension supports this query in version 4 and later.
+
+
+## Device Information Descriptors ##
+
+| Device Descriptors | Return Type | Description |
+| ------------------ | ----------- | ----------- |
+| info\:\:device\:\:ext\_intel\_device\_info\_free\_memory | uint64\_t| Returns the memory avialble on the device in units of bytes.|
```

Comment:
This name seems redundant because "device" and "info" are repeated.  Currently, the name of the descriptor is like:

```
sycl::info::device::ext_intel_device_info_free_memory
```

It seems like we could shorten this to:

```
sycl::info::device::ext_intel_free_memory
```

Actually, you should also coordinate with Rana Rauf, who is changing the namespace of these info descriptors.  After that change the descriptor would be like:

```
sycl::ext::intel::info::device::free_memory
```

(Essentially, we use the `sycl::ext::intel` namespace instead of the `ext_intel_` prefix.)

I'm not sure if it's better for you to add this descriptor with the old name, and then let Rana change it to the new style vs. you adding this new descriptor now with the new style.  Either way is OK with me as long as it ends up with the new style.

Reply:
I thought about this too, but then the the `dev.has(aspect::ext_intel_free_memory)` could become ambiguous since it wouldn't have `device` and `info` in it. Today we have a mixture of everything and I am not adding more mess. Can change to anything that we define as a standard.

Reply:
It looks like all of the existing queries in this spec have a consistent naming except for the "uuid" query.  Those queries omit both "info" and "device" from the name of the descriptor and from the name of the aspect.  Following that style, you would have:

```
if (dev.has(aspect::ext_intel_free_memory)) {
  auto FreeMemory = dev.get_info<ext::intel::info::device::free_memory>();
}
```

I agree that there could be some confusion about the aspect name.  We could add the word "info" to the aspect (only) to avoid this confusion:

```
if (dev.has(aspect::ext_intel_info_free_memory)) {
  auto FreeMemory = dev.get_info<ext::intel::info::device::free_memory>();
}
```

There is no confusion about "device" because all aspects refer to a device.

If we like that naming style, we can change the other aspect names to match, leaving the existing ones as deprecated.  We could also update the "uuid" query to follow this style.

Reply:
I don't feel that strongly, so let's match the style of the majority of the existing queries and use this style:

```
if (dev.has(aspect::ext_intel_free_memory)) {
  auto FreeMemory = dev.get_info<ext::intel::info::device::free_memory>();
}
```